src/ds3_net.h

```diff
@@ -38,6 +38,7 @@ typedef struct {
 
 char* escape_url(const char* url);
 char* escape_url_object_name(const char* url);
+char* escape_url_range_header(const char* url);
```

Comment:
I would write some unit tests that just test this method.  And throw a bunch of weird characters at it.